lib/pages/survey/survey_question_page.dart

```diff
@@ -2,111 +2,147 @@ import 'package:cached_network_image/cached_network_image.dart';
-                          children: [
-                            Text(
-                              "TODO: 1/5",
-                              style: Theme.of(context)
-                                  .textTheme
-                                  .bodyText1
-                                  .copyWith(
-                                      color: Colors.white70, fontSize: 15),
-                            ),
-                            Text(
-                              "TODO: How satisfied you are?",
-                              style: Theme.of(context).textTheme.headline4,
-                              maxLines: 4,
-                              overflow: TextOverflow.ellipsis,
-                            ).marginOnly(top: 8),
-                          ],
-                        ),
-                      ],
-                    ),
-                  ),
+                  _buildBackground(controller.currentQuestion),
+                  _buildTitle(context, controller.indexString,
+                      controller.currentQuestion.text),
+                  _buildAnswerBody(controller.optionalSurvey.value.questions),
+                  _buildSubmitButton(),
                 ],
-              ),
-            ),
-            // TODO: remove these:
-            Align(
-              alignment: Alignment.center,
-              child: SurveyAnswer(
-                type: "choice", // TODO: Come from API
-                optionsText: [
-                  "Your mum",
-                  "Your dad",
-                  "Magic Mike",
-                  "More",
-                ], // TODO: Come from API
-              ),
+              );
+            });
+          }),
+    );
+  }
+
+  Widget _buildBackground(SurveyQuestion question) {
+    return Stack(
+      children: [
+        Container(
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: CachedNetworkImageProvider(question.hdCoverImageUrl,
+                  cacheKey: question.id),
+              fit: BoxFit.cover,
             ),
-            Align(
-              alignment: Alignment.bottomRight,
-              child: FloatingActionButton(
-                child: const Icon(
-                  Icons.chevron_right,
-                  size: 50,
+          ),
+        ),
+        Container(
+          color: Colors.black54,
+        )
+      ],
+    );
+  }
+
+  Widget _buildExitQuestionButton() {
+    return Align(
+      alignment: Alignment.centerRight,
+      child: MaterialButton(
+        color: Colors.white.withOpacity(0.2),
+        height: 28,
+        child: Icon(
+          Icons.close,
+          size: 18,
+          color: Colors.white,
+        ),
+        onPressed: () {
+          // TODO: update this behavior: show confirmation dialog
+          Get.back();
+        },
+        shape: CircleBorder(),
+      ),
+    );
+  }
+
+  Widget _buildTitle(BuildContext context, String indexString, String text) {
```

Comment:
same here `questionPositionString` 👍

Reply:
I make this becomes:

`Widget _buildTitle(BuildContext context, String indexTitle, String questionTitle)`

so I think this is clearer